pilot/pkg/model/jwks_resolver.go

```diff
@@ -321,10 +322,10 @@ const FakeJwks = `{
 }`
 
 // Resolve jwks_uri through openID discovery.
-func (r *JwksResolver) resolveJwksURIUsingOpenID(issuer string) (string, error) {
+func (r *JwksResolver) resolveJwksURIUsingOpenID(ctx context.Context, issuer string, timeout time.Duration) (string, error) {
```

Comment:
I mean init the ctx with timeout, then we donot need to pass timeout anymore

Reply:
ah sorry. I get it now.
I will perform the changes right away

Reply:
I quickly realized that there may be a problem in initializing context with timeout and getting rid of the separate timeout variable.
Timeout is needed for filling the jwtPubKeyEntry that is used to periodically refresh the pubkey in the background process. I don't think I can fill those entries based on the Context.Deadline alone